test/BodyParams/BodyParamsMiddlewareTest.php

```diff
@@ -77,7 +78,8 @@ private function mockHandlerToNeverTrigger()
         return $handler;
     }
 
-    public function jsonProvider()
+    /** @return array<array-key, string[]> */
```

Comment:
Doesn't phpcs raise errors for this? If it does, make this a `@psalm-return` annotation instead (which is what I've done in other packages).

(Apply elsewhere as necessary; I see it in a number of locations.)

Reply:
phpcs doesn't complain at all and adding a duplicate `@psalm-return` doesn't appear to affect psalm's "score"

Reply:
Good to know - phpcs used to complain, so this means current versions are recognizing this syntax.